### PR TITLE
Remove sessionAffinity ClientIP from services

### PR DESCRIPTION
**Pull request by ejweber**
#### Which issue(s) this PR fixes:

Issue #7399

#### What this PR does / why we need it:

Remove `sessionAffinity: ClientIP` from Longhorn services. See #7399 for justification.

#### Special notes for your reviewer:

We could optionally introduce a Helm variable that controls the value of `sessionAffinity` for the Longhorn services that previously used `sessionAffinity: ClientIP`. If we did this, I would argue for making `None` the default, with the option to change back to `ClientIP` if desired. However:

1. I cannot identify any situation in which we should be using `sessionAffinity: ClientIP`,
2. More Helm variables makes Longhorn deployment more difficult to understand.

I'm curious whether reviewers agree.

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM.

**Review by PhanLe1010 — APPROVED** [review]
LGTM

Agree that we should remove the setting completely!

Quick question. Assume that we have `sessionAffinity: ClientIP` set. In the first time the client connects to the backend service it is routed to manager pod with IP-1. Then manager pod with IP-1 crash, restarted, and get a new IP-2. When client connect to backend service for the 2nd time, will Kubernetes network control plane route the client to a new IP different than the IP-1 assume no manager pod has IP-1 anymore?

**Comment by ejweber** [comment]
> Quick question. Assume that we have sessionAffinity: ClientIP set. In the first time the client connects to the backend service it is routed to pod with IP-1. Then pod with IP-1 crash, restarted, and get new IP-2. When client connect to backend service for the 2nd time, will Kubernetes network control plane route the client to a new IP different than the IP-1 assume no manager pod has IP-1 anymore?

Yes. When the longhorn-manager pods want to communicate with the longhorn-backend service, the DNS lookup always returns the same ClusterIP. So, from the perspective of all longhorn-manager pods, there is only one IP address to communicate with. However, as soon as the Kubernetes network control plane recognizes a change in available longhorn-backend pods, it updates the associated endpoint slice. This causes kube-proxy on every node to update DNAT iptables rules appropriately. The next connection attempt makes use of the updated DNAT rules (with the updated list of pod IPs).

**Comment by ejweber** [comment]
> Quick question. Assume that we have sessionAffinity: ClientIP set. In the first time the client connects to the backend service it is routed to pod with IP-1. Then pod with IP-1 crash, restarted, and get new IP-2. When client connect to backend service for the 2nd time, will Kubernetes network control plane route the client to a new IP different than the IP-1 assume no manager pod has IP-1 anymore?

I think yes, though I am interested enough to test it.

From https://github.com/longhorn/longhorn/issues/7399#issuecomment-1896494098:

---
The chain that handles load balancing (`KUBE-SVC-NQL64TY2T3FWFXFQ`) has two kinds of rules:

- The first set of rules are for packets from source IPs that have already been balanced in the last configured number of seconds (`10800`). If, for example, the source IP is in the `KUBE-SEP-W3Y4QUAIHX5U3K5C` list, we jump to the `KUBE-SEP-W3Y4QUAIHX5U3K5C` chain.
- The second set of rules are for packets from source IPs that have not already been balanced. Probabilities are used to ensure we have an equal chance to jump to any related chain (e.g. `KUBE-SEP-W3Y4QUAIHX5U3K5C`).
---

The event you described should cause kube-proxy to remove the list of source IPs that previously balanced to the old destination IP and the rule that would check that list (e.g. `KUBE-SEP-W3Y4QUAIHX5U3K5C`). The source IP is obviously not in any of the other, still existing, lists, so we would eventually fall down to the set of rules that does the load balancing and DNAT to a different IP address.

In other words, I don't think `sessionAffinity: ClientIP` is likely to be the cause of any previously strange connectivity behavior we may have observed.

**Comment by innobead** [comment]
@mergify backport v1.6.x v1.5.x v1.4.x
